src/main/core/support/configuration.rs

```diff
@@ -16,40 +524,230 @@ pub enum InterposeMethod {
+fn default_some_time_1() -> Option<units::Time<units::TimePrefixUpper>> {
+    Some(units::Time::new(1, units::TimePrefixUpper::Sec))
+}
+
+/// Helper function for serde default `Some(LogLevel::Message)` values.
+fn default_some_message() -> Option<LogLevel> {
+    Some(LogLevel::Message)
+}
+
+const DEFAULT_TOPOLOGY: &str = r#"<?xml version="1.0" encoding="utf-8"?>
+<graphml xmlns="http://graphml.graphdrawing.org/xmlns" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://graphml.graphdrawing.org/xmlns http://graphml.graphdrawing.org/xmlns/1.0/graphml.xsd">
+    <key attr.name="packet_loss"    attr.type="double" for="edge" id="edge_packet_loss" />
+    <key attr.name="jitter"         attr.type="double" for="edge" id="edge_jitter" />
+    <key attr.name="latency"        attr.type="double" for="edge" id="edge_latency" />
+    <key attr.name="type"           attr.type="string" for="node" id="node_type" />
+    <key attr.name="bandwidth_up"   attr.type="string" for="node" id="node_bandwidth_up" />
+    <key attr.name="bandwidth_down" attr.type="string" for="node" id="node_bandwidth_down" />
+    <key attr.name="country_code"   attr.type="string" for="node" id="node_country_code" />
+    <key attr.name="ip"             attr.type="string" for="node" id="node_ip" />
+    <key attr.name="packet_loss"    attr.type="double" for="node" id="node_packet_loss" />
+    <graph edgedefault="undirected">
+        <node id="poi-1">
+            <data key="node_packet_loss">0.0</data>
+            <data key="node_ip">0.0.0.0</data>
+            <data key="node_country_code">XX</data>
+            <data key="node_bandwidth_down">1 Gbit</data>
+            <data key="node_bandwidth_up">1 Gbit</data>
+            <data key="node_type">net</data>
+        </node>
+        <edge source="poi-1" target="poi-1">
+            <data key="edge_latency">1.0</data>
+            <data key="edge_jitter">0.0</data>
+            <data key="edge_packet_loss">0.0</data>
+        </edge>
+    </graph>
+</graphml>"#;
+
+/// Generate help strings for objects in a JSON schema, including the Serde defaults if available.
+fn generate_help_strs(
+    schema: schemars::schema::RootSchema,
+) -> std::collections::HashMap<String, String> {
+    let mut defaults = std::collections::HashMap::<String, String>::new();
+    for (name, obj) in &schema.schema.object.as_ref().unwrap().properties {
+        let meta = obj.clone().into_object().metadata.unwrap();
+        let description = meta.description.or(Some("".to_string())).unwrap();
+        let space = if description.len() > 0 { " " } else { "" };
+        match meta.default {
+            Some(default) => defaults.insert(
+                name.clone(),
+                format!("{}{}[default: {}]", description, space, default),
+            ),
+            None => defaults.insert(name.clone(), description.to_string()),
+        };
+    }
+    defaults
+}
+
+fn tilde_expansion(path: &str) -> std::path::PathBuf {
```

Comment:
Optional - use `shellexpand` crate? https://stackoverflow.com/questions/54267608/expand-tilde-in-rust-path-idiomatically

Reply:
I changed it to use this crate, but then remembered the main reason we didn't want to use it:

> Note: because I no longer have capacity to support it, I'm now looking for a new maintainer for this library. Until I'm able to find one, it is unlikely to receive new updates in any reasonably timely manner.

For now I just left in the existing `tilde_expansion()`, but in the future we can switch it out with a crate if we find one that's being maintained.